account_financial_report_webkit_xls: Port to 8.0

anthony-muschang opened the pull request:
I ported the module "account_financial_report_webkit_xls" to 8.0.

This pull request is based on https://github.com/OCA/reporting-engine/pull/5.

All the modifications in this pull request are the result of the migration of account_financial_report_webkit to v8 (see https://github.com/OCA/account-financial-reporting/pull/20).

anthony-muschang committed:
[CHG]account_financial_report_webkit_xls: Migration to v8

anthony-muschang committed:
[CHG]account_financial_report_webkit_xls: Migration to v8

pedrobaeza commented:
You have to fix PEP8 issue with one line length and modify Travis file to download reporting-engine branch for get the dependencies solved.

anthony-muschang committed:
[FIX]account_financial_report_webkit_xls: pep8 + travis

anthony-muschang committed:
[FIX]account_financial_report_webkit_xls: pep8

anthony-muschang commented:
The pep8 issues are fixed and the travis file is updated.

pedrobaeza commented on account_financial_report_webkit_xls/report/open_invoices_xls.py:
```diff
@@ -725,10 +725,12 @@ def print_group_cumul_account(self, row_position, row_start_account,
 
-        if account.grouped_ledger_lines and account.partners_order:
+        if account.grouped_ledger_lines and \
+           _p['partners_order'][account.id]:
```
Bad indentation. Should be more indented.

pedrobaeza commented on account_financial_report_webkit_xls/report/open_invoices_xls.py:
```diff
@@ -757,13 +759,16 @@ def print_grouped_line_report(self, row_pos, account, _xs, xlwt, _p, data):
 
-        if account.ledger_lines and account.partners_order:
+        if _p['ledger_lines'][account.id] \
+           and _p['partners_order'][account.id]:
```
Bad indentation

pedrobaeza commented:
Why are you using `_p` instead of account variable?

anthony-muschang commented:
> Bad indentation

How the line should be indented? There is no pep8 error. Where can I find the guildline on this subject?

> Why are you using `_p` instead of account variable?

Since `account_financial_report_webkit` was migrated to v8, some data like `partners_order` are not stored as account variables anymore, but are instead stored in `_p`. See the pull request #20 for more information.

pedrobaeza commented:
There is no force in PEP8 standard to do it that way, that's why Travis doesn't complain, but as you can see here, it's advisable to do it, specially when you don't use parenthesis:

http://legacy.python.org/dev/peps/pep-0008/#indentation

But this is not a blocking issue. What I see a problem is to use Luc de Meyer branch in Travis instead of the official one, because we will need to change again when the PR is merged. What we have to do is to speed up https://github.com/OCA/reporting-engine/pull/5, and then, we can put the official one.

I understand the point about _p. Thanks for the explanation.

anthony-muschang committed:
[CHG] travis.yml

Use the official branch 8.0 since https://github.com/OCA/reporting-engine/pull/5 has been merged.

pedrobaeza commented:
Thanks for the review on report_xls PR.

I give you my :+1:

We need to wait for at least another one.

guewen commented:
:+1: 
Thanks
